tests/prediction/fuzzy_test.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import pytest
+from beartype import roar
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+from eis_toolkit.prediction.fuzzy_overlay import and_overlay, gamma_overlay, or_overlay, product_overlay, sum_overlay
+
+RASTER_DATA_1 = np.array([[1.0, 0.5, 0.2], [0.7, 0.6, 0.5], [1.0, 0.2, 0.5]])
+
+RASTER_DATA_2 = np.array([[0.7, 0.3, 0.1], [0.4, 0.9, 0.7], [0.9, 0.2, 1.0]])
+
+RASTERS_DATA = np.stack((RASTER_DATA_1, RASTER_DATA_2))
+
+
+def test_and_overlay():
+    """Test that AND overlay works as expected."""
+    result = and_overlay(data=RASTERS_DATA)
+    assert np.array_equal(result, np.array([[0.7, 0.3, 0.1], [0.4, 0.6, 0.5], [0.9, 0.2, 0.5]]))
+
+
+def test_or_overlay():
+    """Test that OR overlay works as expected."""
+    result = or_overlay(data=RASTERS_DATA)
+    assert np.array_equal(result, np.array([[1.0, 0.5, 0.2], [0.7, 0.9, 0.7], [1.0, 0.2, 1.0]]))
+
+
+def test_product_ovelay():
+    """Test that PRODUCT overlay works as expected."""
+    result = product_overlay(data=RASTERS_DATA)
+    assert np.allclose(result, np.array([[0.7, 0.15, 0.02], [0.28, 0.54, 0.35], [0.9, 0.04, 0.5]]))
+
+
+def test_sum_overlay():
+    """Test that SUM overlay works as expected."""
+    result = sum_overlay(data=RASTERS_DATA)
+    assert np.allclose(result, [[1.0, 0.65, 0.28], [0.82, 0.96, 0.85], [1.0, 0.36, 1.0]])
+
+
+def test_gamma_overlay():
+    """Test that GAMMA overlay works as expected."""
+    result = gamma_overlay(data=RASTERS_DATA, gamma=0.6)
+    assert np.array_equal(
```

Comment:
I see `np.testing.assert_array_equal` often used which does the `assert` within. Might be more suitable for use in `pytest` tests. Might do more fancy reporting of the differences.

```suggestion
    np.testing.assert_array_equal(
```